File: app/src/main/java/org/codepond/imdemo/chat/ChatContracts.java
package org.codepond.imdemo.chat;

import org.codepond.imdemo.BasePresenter;
import org.codepond.imdemo.ChatMessage;

import java.util.List;

interface ChatContracts {
    interface Presenter extends BasePresenter {
        void sendMessage(String message);
        void loadMessages();
    }

    interface View {
        void showMessages(List<ChatMessage> chatMessages);
        void notifyNewMessageAdded();
        void cleanUserInput();
    }
}

File: app/src/main/java/org/codepond/imdemo/chat/ChatContracts.kt
package org.codepond.imdemo.chat

import org.codepond.imdemo.BasePresenter
import org.codepond.imdemo.ChatMessage

interface ChatContracts {
    interface Presenter : BasePresenter {
        fun sendMessage(message: String)
        fun loadMessages()
    }

    interface View {
        fun showMessages(chatMessages: MutableList<ChatMessage>)
        fun notifyNewMessageAdded()
        fun cleanUserInput()
    }
}
